tyr-runner/src/main/java/org/jboss/tyr/ci/TeamCityCI.java

```diff
@@ -101,29 +124,100 @@ public void triggerBuild(JsonObject prPayload) {
+                if (prop.name.equals("take-started-build-with-same-revisions") || prop.name.equals("take-successful-builds-only")) {
+                    prop.value = String.valueOf(retestOnlyFailed);
+                }
+            }
+        }
+
+        Jsonb jsonb = JsonbBuilder.create();
+        String stringJsonb = jsonb.toJson(snapshotDependencies);
+        Entity<String> json = Entity.json(stringJsonb);
+
+        String path = "app/rest/buildTypes/id:" + buildId + "/snapshot-dependencies";
+        WebTarget target = returnTeamCityTarget(path);
+        Response response = null;
+
+        try {
+            response = target.request()
+                .header(HttpHeaders.ACCEPT_ENCODING, "UTF-8")
+                .header(HttpHeaders.CONTENT_TYPE, MediaType.APPLICATION_JSON)
+                .header(HttpHeaders.AUTHORIZATION, "Basic " + encryptedCredentials)
+                .put(json);
+
+        } catch (Throwable e) {
+            log.error("Cannot retrieve JSON on the path " + path, e);
+            throw e;
+        } finally {
+            if (response != null) {
+                response.close();
+            }
+        }
     }
 
-    private Map<String, String> parseBranchMapping(String mappings) {
-        Map<String, String> branchMap = new HashMap<>();
+    private SnapshotDependencies getSnapshotDependencies(String buildId) {
+        String path = "app/rest/buildTypes/id:" + buildId + "/snapshot-dependencies";
+        WebTarget target = returnTeamCityTarget(path);
+        Response response = null;
+        SnapshotDependencies snapshotDependencies;
 
-        for (String mapping : mappings.split(",")) {
-            String[] parts = mapping.split("=>");
-            branchMap.put(parts[0].trim(), parts[1].trim());
+        try {
+            response = target.request()
+                .header(HttpHeaders.ACCEPT_ENCODING, "UTF-8")
+                .header(HttpHeaders.CONTENT_TYPE, MediaType.APPLICATION_JSON)
+                .header(HttpHeaders.AUTHORIZATION, "Basic " + encryptedCredentials)
+                .header(HttpHeaders.ACCEPT, MediaType.APPLICATION_JSON)
+                .get();
+
+            String json = response.readEntity(String.class);
+            Jsonb jsonb = JsonbBuilder.create();
+            snapshotDependencies = jsonb.fromJson(json, SnapshotDependencies.class);
+
+        } catch (Throwable e) {
+            log.error("Cannot retrieve JSON on the path " + path, e);
+            throw e;
+        } finally {
+            if (response != null) {
+                response.close();
+            }
         }
+        return snapshotDependencies;
+    }
 
-        return branchMap;
+    private void setTriggerBuildProperties(JsonObject prPayload) {
+        branch = prPayload.getJsonObject(Utils.BASE).getString(Utils.REF);
+        pull = prPayload.getInt(Utils.NUMBER);
+        sha = prPayload.getJsonObject(Utils.HEAD).getString(Utils.SHA);
+        buildId = branchMappings.get(branch);
```

Comment:
so you are setting these fields on every invocation of triggerRetesting*. This kind of storing values is inefficient. If they are stored only for one method invocation then it is not necessary to store them in fields.